### PR TITLE
feat(flat-money-stablelp): flat money stableLp

### PR DESCRIPTION
**Pull request by kent426**

### CODE REVIEW

File: src/services/flatmoney/stableLp.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { ethers } from "../..";
+import DelayedOrderAbi from "../../abi/flatmoney/DelayedOrder.json";
+
+export function getAnnounceStableDepositTxData(
+  depositAmount: ethers.BigNumber | string,
+  minAmountOut: ethers.BigNumber | string,
+  keeperFee: ethers.BigNumber | string
```
Comment (hochbruj) [inline src/services/flatmoney/stableLp.ts:12]: Do you think we could calculate the minAmountOut internally by just providing a slippage percentage?
Maybe keeperFee could also be derived?

Comment (kent426) [reply src/services/flatmoney/stableLp.ts:14]: yes, keeperFee and slippage are derived now if you look at the pool functions. 9a99688

File: src/services/flatmoney/keeperFee.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+  let keeperfee: ethers.BigNumber;
+  if (basefee) {
+    keeperfee = await keeperFeeContract["getKeeperFee(uint256)"](
+      new BigNumber(basefee.toString()).times(2).toFixed(0)
+    );
+  } else {
+    keeperfee = await keeperFeeContract["getKeeperFee()"]();
+  }
+
```
Comment (hochbruj) [inline src/services/flatmoney/keeperFee.ts:49]: Is it this how the keeperfee is calculated in UI? We should just have the same calculation.

**Review by hochbruj — COMMENTED** [review]
I think it is not necessary to check the value of the keeper fee in USD inside of the SDK. We should replicate the logic of the UI. I tested it there and if the input amount in rETH is lower or equal the keeper fee you get an error message. So I would just  do a check inside  of mintUnitViaFlatMoney and throw an error.

File: src/services/flatmoney/keeperFee.ts (hunk 2 above)
Comment (hochbruj) [reply src/services/flatmoney/keeperFee.ts:49]: I checked the frontend code,  they are passing the gasPrice from wagmi to the function and then multiplying with 1.2. I am not sure if the gas price is the base fee from the provider??
https://github.com/dhedge/flatcoin-frontend/blob/0f2830dbf62c1f3da6ba7a8476dea71a3302f48a/libs/flatcoin/page/src/hooks/state/useContractKeeperFee.ts

Comment (kent426) [reply src/services/flatmoney/keeperFee.ts]: and i think it should be ok to keep the maxKeeperFee check.

**Review by hochbruj — APPROVED** [review]

**Pull request merged**